sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -228,6 +225,15 @@ def _infer_event_timestamp_from_dataframe(entity_df: pandas.DataFrame) -> str:
             )
 
 
+class BigQueryOfflineJob(OfflineJob):
+    def __init__(self, query: str, config: RepoConfig):
+        self.query = query
+        self.client = _get_bigquery_client(project=config.offline_store.project_id)
+
+    def to_table(self) -> pyarrow.Table:
+        return self.client.query(self.query).to_arrow()
```

Comment:
Is it intentional for `BigQueryOfflineJob.to_arrow()` to return a pyarrow table? I would have expected `to_table()` to create a BQ table. Would `to_arrow_table()` or `to_arrow()` be clearer?